src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -24,25 +24,179 @@ Copyright (c) .NET Foundation. All rights reserved.
+                 Properties="ComposeWorkingDir=$(ComposeWorkingDir);
+                             PublishDir=$(PublishDir);
+                             CacheStagingDir=$(CacheStagingDir);
+                             TargetFramework=$(_TFM);
+                             JitPath=$(JitPath);
+                             Crossgen=$(Crossgen);
+                             RestoreProj=$(RestoreProj);
+                             DisableImplicitFrameworkReferences=true;
+                             SkipUnchangedFiles=$(SkipUnchangedFiles);
+                             PreserveCacheLayout=$(PreserveCacheLayout)"
+                 ContinueOnError="WarnAndContinue">
+      <Output ItemName="AllResolvedPackagesPublished" TaskParameter="TargetOutputs" />
+    </MSBuild>
+  </Target>
+  <!--
+    ============================================================
+                                        CacheWorkerMapper
 
+   Processes each package specified in a cache project file
+    ============================================================
+    -->
+
+  <Target Name="CacheWorkerMapper"
+          Returns ="@(ResolvedPackagesFromMapper)">
+
+    <ItemGroup>
+      <PackageReferencesToCache Include="$(RestoreProj)">
+        <Properties>
+          CachePackageName=%(PackageReference.Identity);
+          CachePackageVersion=%(PackageReference.Version);
+          ComposeWorkingDir=$(ComposeWorkingDir);
+          PublishDir=$(PublishDir);
+          CacheStagingDir=$(CacheStagingDir);
+          TargetFramework=$(TargetFramework);
+          JitPath=$(JitPath);
+          Crossgen=$(Crossgen);
+          SkipUnchangedFiles=$(SkipUnchangedFiles);
+          PreserveCacheLayout=$(PreserveCacheLayout)
+        </Properties>
+      </PackageReferencesToCache>
+    </ItemGroup>
+
+    <MSBuild Projects="@(PackageReferencesToCache)"
+                 Targets="CacheWorker"
+                 BuildInParallel="$(BuildInParallel)"
+                 ContinueOnError="WarnAndContinue">
+      <Output ItemName="ResolvedPackagesFromMapper" TaskParameter="TargetOutputs" />
+    </MSBuild>
+  </Target>
+
+  <Target Name="CacheWorker"
+          Returns="@(ResolvedPackagesPublished)"
+          DependsOnTargets="PrepforRestoreForComposeCache;
+                            CacheWorkerPerformWork"/>
+  
+  <Target Name="CacheWorkerPerformWork"
+          DependsOnTargets="RestoreForComposeCache;
+                            ComputeAndCopyFilesToCacheDirectory;"
+          Condition="!Exists($(CacheWorkerWorkingDir))" />
+
+<!--
+    ============================================================
+                                        CacheFinalizer
+
+   Cleans up and produces artifacts after completion of cache
+    ============================================================
+    -->
+  
+  <Target Name="CacheFinalizer"
+          DependsOnTargets="CacheWorkerMain;
+                            _CopyResolvedOptimizedFiles">
+    
+    <RemoveDuplicates
```

Comment:
great catch, will remove this logic for now. As i am handling duplicates when consuming them